res/controllers/common-controller-scripts.js

```diff
@@ -9,728 +9,618 @@
+     * This function is useful for creating a soft start effect on a deck, gradually increasing its playback speed to the rate determined by the pitch slider.
+     * @param {number} channel not used
+     * @param {number} control not used
+     * @param {number} value Positive value activates spinback
+     * @param {number} status The status of the MIDI message, if the MSB bit 0x80 of the byte is not set, value is evaluated.
+     * @param {string} group The group where the deck number is derived from, e.g. "[Channel1]"
+     * @param {number} [factor] Multiplier for the acceleration effect. Higher values increase the acceleration of the soft start effect.
+     */
+    softStart(channel, control, value, status, group, factor = 1) {
+        // disable on note-off or zero value note/cc, use default increase rate '1'
+        engine.softStart(
+            parseInt(group.substring(8, 9)), ((status & 0xF0) !== 0x80 && value > 0),
+            factor
+        );
     }
-    this.Buttons[buttonName] = button;
-};
-
-Controller.prototype.setControlValue = function(control, value) {
-    this.Controls[control].setValue(this.group, value);
-};
-
-// Button
 
-var Button = function(controlId) {
-    this.controlId = controlId;
-    this.state = ButtonState.released;
-};
-Button.prototype.handleEvent = function(value) {
-    this.handler(value);
-};
+});
 
-// Control
-
-var Control = function(mappedFunction, softMode) {
-    // These defaults are for MIDI controllers
-    this.minInput = 0;
-    this.midInput = 0x3F;
-    this.maxInput = 0x7F;
-    // ----
-
-    this.minOutput = -1.0;
-    this.midOutput = 0.0;
-    this.maxOutput = 1.0;
-    this.mappedFunction = mappedFunction;
-    this.softMode = softMode;
-    this.maxJump = 10;
-};
 
-Control.prototype.setValue = function(group, inputValue) {
-    let outputValue = 0;
-    if (inputValue <= this.midInput) {
-        outputValue = this.minOutput
-            + ((inputValue - this.minInput) / (this.midInput - this.minInput))
-            * (this.midOutput - this.minOutput);
-    } else {
-        outputValue = this.midOutput
-            + ((inputValue - this.midInput) / (this.maxInput - this.midInput))
-            * (this.maxOutput - this.midOutput);
-    }
-    if (this.softMode) {
-        const currentValue = engine.getValue(group, this.mappedFunction);
-        let currentRelative = 0.0;
-        if (currentValue <= this.midOutput) {
-            currentRelative = this.minInput
-                + ((currentValue - this.minOutput) / (this.midOutput - this.minOutput))
-                * (this.midInput - this.minInput);
-        } else {
-            currentRelative = this.midInput
-                + ((currentValue - this.midOutput) / (this.maxOutput - this.midOutput))
-                * (this.maxInput - this.midInput);
+/**
+ * Used for tapping the desired BPM for a deck
+ */
+// @ts-ignore Same identifier for class and instance needed for backward compatibility
+class bpmClass {
```

Comment:
```suggestion
class bpmClass {
```

I also just noticed that we now have `[ChannelN],tempo_tap` which essentially deprecates `script.bpm`, right? cc @ronso0